src/lib/x509/certstor_flatfile/certstor_flatfile.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+* Certificate Store
+* (C) 1999-2019 Jack Lloyd
+* (C) 2019      Patrick Schmidt
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/build.h>
+
+#include <botan/certstor_flatfile.h>
+#include <botan/data_src.h>
+#include <botan/pem.h>
+
+namespace Botan {
+namespace {
+std::vector<secure_vector<uint8_t>> decode_all(DataSource& source)
+   {
+   std::vector<secure_vector<uint8_t>> pems;
```

Comment:
I know PEM_Code::decode only returns `secure_vector` (which should be fixed), but zeroization/locking is not needed here since the certs are public information. So return `std::vector<std::vector<uint8_t>>` and use `pems.push_back(unlock(PEM_Code::Decode(source, label))`

Also verify the PEM label is one of "CERTIFICATE" or "X509 CERTIFICATE" or "TRUSTED CERTIFICATE".